src/storage/src/hummock/compactor/compactor_runner.rs

```diff
@@ -876,10 +886,12 @@ where
             // delete key to represent this.
             iter_key.epoch_with_gap =
                 EpochWithGap::new_from_epoch(earliest_range_delete_which_can_see_iter_key);
-            sst_builder
-                .add_full_key(iter_key, HummockValue::Delete, is_new_user_key)
-                .verbose_instrument_await("add_full_key_delete")
-                .await?;
+            if !compaction_filter.should_delete(iter_key) {
```

Comment:
I don't get why we need this line. If `iter_key` is filtered by `compaction_filter`, I think it will hit L846 and won't reach here.

Reply:
fixed